darts/utils/data/tabularization.py

```diff
@@ -0,0 +1,136 @@
+from typing import Optional, Sequence, Union
+
+import numpy as np
+import pandas as pd
+
+from darts.logging import raise_if
+from darts.timeseries import TimeSeries
+
+
+def create_lagged_data(
+    target_series: Union[TimeSeries, Sequence[TimeSeries]],
+    output_chunk_length: int,
+    past_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+    future_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+    lags_list: Optional[Sequence[int]] = None,
+    lags_past_covariates_list: Optional[Sequence[int]] = None,
+    lags_future_covariates_list: Optional[Sequence[int]] = None,
+    max_samples_per_ts: Optional[int] = None,
+    is_training: Optional[bool] = True,  # other option: 'inference
+):
+    """
+    Helper function that creates training/validation matrices (X and y as required in sklearn), given series and
+    max_samples_per_ts.
+
+    X has the following structure:
+    lags_target | lags_past_covariates | lags_future_covariates
+
+    Where each lags_X has the following structure (lags_X=[-2,-1] and X has 2 components):
+    lag_-2_comp_1_X | lag_-2_comp_2_X | lag_-1_comp_1_X | lag_-1_comp_2_X
+
+    y has the following structure (output_chunk_length=4 and target has 2 components):
+    lag_+0_comp_1_target | lag_+0_comp_2_target | ... | lag_+3_comp_1_target | lag_+3_comp_2_target
```

Comment:
It would be nice to document each of the parameters (even if, assuming we make the method private, it won't appear in the documentation pages)